### PR TITLE
@reach/checkbox: For review

### PR DESCRIPTION
**Pull request by chancestrickland**
Thank you for contributing to Reach UI! Please fill in this template before submitting your PR to help us process your request more quickly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code (Compile and run).
- [ ] Add or edit tests to reflect the change (Run with `yarn test`). 😢
- [x] Add or edit Storybook examples to reflect the change (Run with `yarn start`).
- [x] Ensure formatting is consistent with the project's Prettier configuration.

This pull request:

- [x] Creates a new package
- [ ] Fixes a bug in an existing package
- [ ] Adds additional features/functionality to an existing package
- [ ] Updates documentation or example code
- [ ] Other

If creating a new package:

- [ ] Make sure the new package directory contains each of the following, and that their structure/formatting mirrors other related examples in the project:
  - [x] `examples` directory
  - [x] `src` directory with an `index.js` entry file
  - [ ] Type definitions in an `index.d.ts` file are desired but not required for the PR to be merged
  - [ ] At least one example file per feature introduced by the new package **WIP**
  - [x] Base styles in a `style.css` file (if needed by the new package)

### CODE REVIEW

**Comment by chancestrickland** [comment]
TODO: Tests, docs, and a few more examples

**Comment by chancestrickland** [comment]
We are likely going to break this into two smaller composable components, `MixedCheckbox` and `CustomCheckbox` and encourage users to use standard HTML inputs for typical boolean logic needs.

File: website/src/pages/dialog.mdx [hunk 1]
```diff
@@ -494,22 +494,22 @@ function Example(props) {
       <p>This might take a while! Are you sure?</p>
       <button>Go Forward</button> <button>Go Back</button>
     </Dialog>
-  )
+  );
```
Comment (chancestrickland) [inline website/src/pages/dialog.mdx]: I'm blaming prettier for this one sneaking in 🤦‍♂️

File: packages/checkbox/examples/group.example.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
```
Comment (ryanflorence) [inline packages/checkbox/examples/group.example.js]: Let's make CheckboxGroup an example for now. Great candidate for Reach DS.

**Pull request closed without merging**